icu4c/source/tools/icuwriteuprops/icuwriteuprops.cpp

```diff
@@ -129,17 +163,36 @@ int main(int argc, char* argv[]) {
         argc=-1;
     }
 
-    if(argc<0 || options[OPT_HELP_H].doesOccur || options[OPT_HELP_QUESTION_MARK].doesOccur) {
+    std::vector<const char*> propNames;
+    for (int i=1; i<argc; i++) {
+        propNames.push_back(argv[i]);
+    }
+    if (options[OPT_ALL].doesOccur) {
+        for (int i=UCHAR_BINARY_START; i<UCHAR_INT_LIMIT; i++) {
+            if (i == UCHAR_BINARY_LIMIT) {
+                i = UCHAR_INT_START;
+            }
+            UProperty uprop = static_cast<UProperty>(i);
+            const char* propName = u_getPropertyName(uprop, U_LONG_PROPERTY_NAME);
```

Comment:
Currently ICU4X uses the short name for the .toml file, so that we can determine the name of the file directly from the key. We can work around that if you think the long name is better for the download, though.

Reply:
I used the long name because I realized that there are a few properties (the POSIX properties "alnum", "blank", etc) that don't have a short name.

Reply:
I recently stumbled on that, too. They should probably have a non-empty short name, and it should be the same as the long name -- like for other properties where we didn't invent unique short names. Care to submit an ICU ticket?

Reply:
In the meantime, that should not stop you from using the short name *where available*.

Reply:
I was a little worried about using the short name and then falling back to the long name, because then the paths within the zip file might change between versions, but this should not be a major concern because code that looks up a particular property in the zip file should be using the `[binary_property].short_name` property in the TOML files.

So I have changed this to create files with the short name by default.